doc/tutorials/dune-package-management/setup.md

```diff
@@ -0,0 +1,197 @@
+possible. Thus most projects can easily be built with just the minimum of
+changes.
+
+In this tutorial we'll create a simple project to use the OCaml package
+management for the very first time.
+
+## Declare dependencies
+
+The best way to work with the package management is to declare your
+dependencies in the `dune-project` file.
+
+::::{dropdown} `dune-project`
+:icon: file-code
+
+:::{literalinclude} setup/dune-project
+:language: dune
+:emphasize-lines: 6-7
+:::
+
+We define a project called `test` and define that to build it we need an OCaml
+compiler newer than 4.14.
+
+This configuration will be very familiar to people who use Dune to generate
+their OPAM files. This is exactly the same information that is used to
+generate the files using the `generate_opam_files` stanza.
+
+::::
+
+::::{dropdown} `test.ml`
+:icon: file-code
+
+:::{literalinclude} setup/test.ml
+:language: ocaml
+:::
+
+To show that the build works, this simple program will be built and executed.
+
+::::
+
+
+::::{dropdown} `dune`
+:icon: file-code
+
+:::{literalinclude} setup/dune
+:language: dune
+:::
+
+To declare our module an executable we need a little bit of configuration; we
+just define the module as an executable.
+
+::::
+
+After our project skeleton is set-up, we can proceed to the next step.
+
+## Locking dependencies
+
+After declaring the dependencies, you will need to tell Dune which package
+versions to use for your project. This is done by creating a lock directory.
+This is easily done with a new Dune command:
+
+```sh
+$ dune pkg lock
+Solution for dune.lock:
+- ocaml.5.2.0
+- ocaml-base-compiler.5.2.0
+- ocaml-config.3
+```
+
+This will update all the required OPAM repositories, use the newest version of
+each and try to find a set of packages and versions that satisfy the
+constraints that your project dependencies declare.
+
+:::{note}
+The versions that get locked might be different from this tutorial, as we only
+specified the lower bound of `ocaml`; bar any additional configuration Dune
+will pick the newest possible version for each dependency.
+:::
+
+## Build project
+
+The build the project you can just use the regular Dune commands to build your
+project.
+
+```sh
+dune build
+```
+
+This will download, build and install all your locked dependencies and then use
+those to build your project. This means that the first time building will take
+longer than usual, as the dependencies need to be built first. Subsequent
+builds where no dependency needs to be updated will be just as fast as before.
+
+We can show that the package has been built successfully and works as expected:
+
+```sh
+$ dune exec ./test.exe
+Hello, OCaml, Rust!
+```
+
+## Adding libraries
+
+Much like in regular projects to add a library we need to add a dependency to
+it. For simplicity we'll use the popular `fmt` library as an example, but any
+library on the OPAM repository can be used.
+
+First we update the `dune-project` file to add a dependeny on the OPAM package.
```

Comment:
Do we need to mention opam here? Isn't it confusing that we are learning how to get rid of opam but still talk about opam? :confused:

Reply:
I think `opam` and OPAM repository and OPAM packages are different things:

  1. `opam` is the tool to install OPAM packages. I'd prefer not to use it since the tutorial shouldn't require users having access to an `opam` binary.
  2. OPAM packages are packages in the OPAM repository. I could call them "packages" or "OCaml packages" but I think "OPAM package" is for better or worse a common way to refer to them.
  3. OPAM repository is here to stay, we use it internally so if I were to call it "package repository" it would probably cause more confusion than improve things.

Reply:
It makes sense! Thanks for the clarification :+1: